src/Ybus.jl

```diff
@@ -842,93 +884,230 @@ function _make_subnetwork_axes(ybus, bus_numbers_to_remove, arcs_to_remove)
-            1 / series_impedance_to_from
-        data[bus_lookup[equivalent_arc[1]], bus_lookup[equivalent_arc[1]]] +=
-            shunt_admittance / 2
-        data[bus_lookup[equivalent_arc[2]], bus_lookup[equivalent_arc[2]]] +=
-            shunt_admittance / 2
+        ordered_bus_numbers, segment_orientations =
+            _get_chain_data(equivalent_arc, series_map_entry, nrd)
+        ordered_bus_indices = [bus_lookup[x] for x in ordered_bus_numbers]
+        equivalent_arc_indices = (ordered_bus_indices[1], ordered_bus_indices[end])
+        ybus_isolated_d2_chain = _build_chain_ybus(series_map_entry, segment_orientations)
+        ybus_boundary_isolated_d2_chain = _reduce_internal_nodes(ybus_isolated_d2_chain)
+        _apply_d2_chain_ybus!(
+            data,
+            ybus_isolated_d2_chain,
+            ybus_boundary_isolated_d2_chain,
+            equivalent_arc_indices,
+        )
+    end
+    return yft, ytf
+end
+
+function _add_series_branches_to_ybus!(
+    data::SparseArrays.SparseMatrixCSC{ComplexF32, Int64},
+    bus_lookup::Dict{Int, Int},
+    yft::ArcAdmittanceMatrix,
+    ytf::ArcAdmittanceMatrix,
+    series_branch_map::Dict{Tuple{Int, Int}, Vector{Any}},
+    nrd::NetworkReductionData,
+)
+    arc_lookup = get_arc_lookup(yft)
+    arc_axis = get_arc_axis(yft)
+    I_yft, J_yft, V_yft = SparseArrays.findnz(yft.data)
+    I_ytf, J_ytf, V_ytf = SparseArrays.findnz(ytf.data)
+    row_ix = size(yft)[1] + 1
+    n_buses = size(yft)[2]
+    for (equivalent_arc, series_map_entry) in series_branch_map
+        ordered_bus_numbers, segment_orientations =
+            _get_chain_data(equivalent_arc, series_map_entry, nrd)
+        ordered_bus_indices = [bus_lookup[x] for x in ordered_bus_numbers]
+        equivalent_arc_indices = (ordered_bus_indices[1], ordered_bus_indices[end])
+        ybus_isolated_d2_chain = _build_chain_ybus(series_map_entry, segment_orientations)
+        ybus_boundary_isolated_d2_chain = _reduce_internal_nodes(ybus_isolated_d2_chain)
+        _apply_d2_chain_ybus!(
+            data,
+            ybus_isolated_d2_chain,
+            ybus_boundary_isolated_d2_chain,
+            equivalent_arc_indices,
+        )
+
+        push!(arc_axis, equivalent_arc)
+        push!(I_yft, row_ix)
+        push!(I_yft, row_ix)
+        push!(J_yft, equivalent_arc_indices[1])
+        push!(J_yft, equivalent_arc_indices[2])
+        push!(V_yft, ybus_boundary_isolated_d2_chain[1, 1])
+        push!(V_yft, ybus_boundary_isolated_d2_chain[1, 2])
+        push!(I_ytf, row_ix)
+        push!(I_ytf, row_ix)
+        push!(J_ytf, equivalent_arc_indices[2])
+        push!(J_ytf, equivalent_arc_indices[1])
+        push!(V_ytf, ybus_boundary_isolated_d2_chain[2, 2])
+        push!(V_ytf, ybus_boundary_isolated_d2_chain[2, 1])
+        row_ix += 1
     end
+    yft_data = SparseArrays.sparse(I_yft, J_yft, V_yft, row_ix - 1, n_buses)
+    ytf_data = SparseArrays.sparse(I_ytf, J_ytf, V_ytf, row_ix - 1, n_buses)
```

Comment:
Can you add a comment why you add a + 1 offset at the top loop and then remove it when making the sparse object?